tools/crashlytics/defs.bzl

```diff
@@ -0,0 +1,69 @@
+load("@tools_android//tools/googleservices:defs.bzl", "google_services_xml")
+
+def crashlytics_android_library(
+    name,
+    package_name,
+    build_id,
+    google_services_json):
+
+  _CRASHLYTICS_PROP_TEMPLATE = \
+  """build_id={build_id}
+package_name={package_name}"""
+  crashlytics_properties_file = "crashlytics-build.properties"
+  crashlytics_properties_file_content = _CRASHLYTICS_PROP_TEMPLATE.format(
+      build_id = build_id,
+      package_name = package_name,
+  )
+
+  native.genrule(
+      name = "%s_crashlytics_setup_properties" % name,
+      outs = [crashlytics_properties_file],
+      tools = ["@tools_android//tools/crashlytics"],
+      cmd = "$(location @tools_android//tools/crashlytics) \"%s\" $@" % crashlytics_properties_file_content,
+  )
+
+  # Generate the unique identifier for Fabric backend to identify builds.
+  # https://docs.fabric.io/android/crashlytics/build-tools.html#optimize-builds-when-you-re-not-proguarding-or-using-beta-by-crashlytics
+  _CRASHLYTICS_RES_TEMPLATE = \
+  """<?xml version=\\"1.0\\" encoding=\\"utf-8\\" standalone=\\"no\\"?>
+<resources xmlns:tools=\\"http://schemas.android.com/tools\\">
+    <string tools:ignore=\\"UnusedResources,TypographyDashes\\" name=\\"com.crashlytics.android.build_id\\" translatable=\\"false\\">{build_id}</string>
+</resources>"""
+  crashlytics_res_values_file = "res/values/com_crashlytics_build_id.xml"
+  crashlytics_res_values_file_content = _CRASHLYTICS_RES_TEMPLATE.format(build_id = build_id)
+
+  native.genrule(
+      name = "%s_crashlytics_setup_res" % name,
+      outs = [crashlytics_res_values_file],
+      tools = ["@tools_android//tools/crashlytics"],
+      cmd = "$(location @tools_android//tools/crashlytics) \"%s\" $@" % crashlytics_res_values_file_content,
+  )
+
+  _CRASHLYTICS_MANIFEST_TEMPLATE = \
+"""<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
+<manifest xmlns:android=\\"http://schemas.android.com/apk/res/android\\"
+          package=\\"{package_name}\\">
+</manifest>
+"""
+  crashlytics_manifest_file = "CrashlyticsManifest.xml"
+  crashlytics_manifest_file_content = _CRASHLYTICS_MANIFEST_TEMPLATE.format(package_name = package_name)
+
+  native.genrule(
+      name = "%s_crashlytics_setup_manifest" % name,
+      outs = [crashlytics_manifest_file],
+      tools = ["@tools_android//tools/crashlytics"],
+      cmd = "$(location @tools_android//tools/crashlytics) \"%s\" $@" % crashlytics_manifest_file_content,
+  )
+
+  google_services_xml_files = google_services_xml(
```

Comment:
so google_services_xml will be a noop if there's an existing google_services_xml, but it would probably be better to have the user specify this separately from crashlytics (say the only google_service_xml is here, and they remove crashlytics. other things like FCM would fail)

Reply:
Great point. I've extracted google_services_xml out from crashlytics_android_library and added a resource_files parameter to crashlytics_android_library to pull in the generated XML. The motivation is to make the dependency of crashlytics on the generated XML clearly, and let the crashlytics_android_library be self-contained.

Reply:
The final BUILD file will look like this:

```python
GOOGLE_SERVICES_RESOURCES = google_services_xml(
    package_name = "com.example.package",
    google_services_json = "google-services.json",
)
crashlytics_android_library(
    name = "crashlytics_lib",
    package_name = "com.example.package",
    build_id = "9dfea8fe-4d75-48a7-ba28-4ddb7fe74780",
    resource_files = GOOGLE_SERVICES_RESOURCES,
)
android_library(
    name = "crashlytics_deps",
    exports = [
        artifact("com.crashlytics.sdk.android:crashlytics"),
        artifact("io.fabric.sdk.android:fabric"),
    ],
)
```